src/server/server_family.cc

```diff
@@ -461,7 +461,8 @@ ServerFamily::~ServerFamily() {
 void SetMaxClients(std::vector<facade::Listener*>& listeners, uint32_t maxclients) {
   for (auto* listener : listeners) {
     if (!listener->IsPrivilegedInterface()) {
-      listener->SetMaxClients(maxclients);
+      listener->socket()->proactor()->Await(
```

Comment:
This was a data race between the proactor that runs the listener and reads this variable and the proactor that handles the Invocation of this config set command